### ISSUE TITLE
Nachbarschaftsdomänen in Karte verlinken

### ISSUE THREAD

**Issue by MPW1412** [FreiFunkMuenster/md-fw-dl#11]
Moin,

um die Aufstellung von Knoten in Nachbarcommunities einzudämmen, hatten wir vor einiger Zeit mal die Idee, noch Polygone in die Karte einzubauen, die an Freifunk Münsterland grenzen und dann eine Weiterleitung zu deren Firmwaredownloadseiten zu machen.

Grüße
Matthias

**Comment by descilla** [comment]
Dazu habe ich noch zwei offene Punkte:
- Welches Verhalten wird erwartet? Was soll genau passieren, wenn man auf das Nachbar-Shape klickt?
- Wie sollen die Shapes der Nachbarcommunities aussehen? Angrenzender Kreis? Angrenzendes Bundesland? Ich bin da noch nicht auf eine brauchbare Idee gekommen.

**Comment by MPW1412** [comment]
Als Verhalten würde ich vorschlagen, dass wir - je nach Wunsch der Nachbarcommunity - direkt auf die Firmware verlinken, oder nur auf die Website. Ich würde erstmal die Website nehmen.

Bzgl. Größe, der Nachbarkreis klingt gut. Das sollte grob passen und wäre dank OSM nicht zu viel Aufwand; außer im Norden, aber da kann man eventuell über die Differenzmenge zum jetzigen Polygon arbeiten.

**Comment by christian-weiss** [comment]
Dagegen spricht, dass der Firmware Download Assistent schon jetzt reichlich HTTP-Requests absetzt um einsatzbereit zu werden (aktuell: 74 Domänen-Shape-Files bei 115 Requests gesamt). Mit Nachbar-Shapes kommen vermutlich noch mal 30-50 hinzu.

Auf einer langsamen Verbindung oder bei hohen Latenzen fühlt sich der Assistent schon jetzt nicht mehr "State of the art" an. Und ich finde, die "schlecht angebundenen" sollten uns ganz besonders am Herzen liegen.

Vielleicht wäre ein einziger "Kranz-Shape" der sich um das Kerngebiet anschmiegt ein Kompromiss - einfach ein Link auf eine statische Info Seite, wo alle Nachbar-Communities gelistet sind. Aber wer hat wirklich Lust die ganzen Links aktuell zu halten? Ein Crawling der Community API gemäß einer Nachbarschaftliste könnte den Aufwand reduzieren...

Alternativ könnte man auch alle Shape-Files durch ein build-script in einer Datei zusammenfassen, dazu müssen aber auch das JavaScript etwas angepasst werden. Und das Download-Volumen sowie das JavaScript-Rendering könnte man durch Polygon-Vereinfachung um den Faktor 10 reduzieren sowie das JavaScript-Parsing durch Minifizierung beschleunigen (reduziert auch das Volumen ein bischen).

**Comment by MPW1412** [comment]
Bzgl. der vielen Verbindungen müssen wir auf dem Webserver einfach mal HTTP/2 aktivieren. Dann werden die automatisch zusammengefasst.

Die geojsons zusammen zu fassen fände ich auch interessant. Gibt es da einen Ansatz?

**Comment by descilla** [comment]
> Bzgl. der vielen Verbindungen müssen wir auf dem Webserver einfach mal HTTP/2 aktivieren. Dann werden die automatisch zusammengefasst.
> 
> Die geojsons zusammen zu fassen fände ich auch interessant. Gibt es da einen Ansatz?

Natürlich. Einzelne Dateien für die Domains wurde seinerzeit gemacht, als wir 6 Domänen hatten. Und ich dachte, dass es einfacher sei die Daten (ist ja letztendlich ein Export aus OSM) zu verwalten.

Man kann natürlich einfach alle Shapes in eine geojson-Datei schmeißen. Man müsste dann hier diese Schleife umbauen: https://github.com/FreiFunkMuenster/md-fw-dl/blob/master/map.js#L87:L99 Anstatt über alle Domänen zu iterieren und das jeweilige geojson zu laden, lädt man einmal das geojson mit allen shapes und iteriert dann da drüber.

Man muss vermutlich nur an jedes Shape ein Attribut packen, um es einer entsprechenden Domäne zuzuordnen.

**Comment by descilla** [comment]
Nachtrag: Ich würde die Shapes vermutlich weiterhin in einzelnen Dateien speichern und dann mittels Script vom Build-Server in eine Datei schreiben lassen. Dann kann man auch direkt das passende Attribut dran pappen.
